src/components/modals/Stake.tsx

```diff
@@ -79,7 +79,7 @@ export const StakeModal: FC<IStakeModalProps> = ({
 	);
 
 	useEffect(() => {
-		setPermit(!onlyApproveMode);
+		handlePermit();
```

Comment:
It's wrong, it switches to permit mode even if it's not allowed!
```suggestion
		setPermit(!onlyApproveMode);
		setStakeState(StakeState.APPROVE);
```